src/XrdHttp/README-CKSUM.md

```diff
@@ -0,0 +1,15 @@
+In the case XrdHttp has been configured, it is important that the configuration contains at least one digest algorithm registered in the IANA database: https://www.iana.org/assignments/http-dig-alg/http-dig-alg.xhtml
+Otherwise the user will get a 403 error for each checksum request they will do.
+
+Here is a table summarizing the IANA-compliant checksum names:
+
+| Digest algorithm | HTTP digest | Will be base64 encoded |
+|------------------|-------------|------------------------|
+| md5              | md5         | true                   |
+| adler32          | adler32     | false                  |
+| sha1             | sha         | true                   |
+| sha256           | sha-256     | true                   |
+| sha512           | sha-512     | true                   |
+| cksum            | UNIXcksum   | false                  |
```

Comment:
This is the IANA definition:

https://www.iana.org/assignments/http-dig-alg/http-dig-alg.xhtml

> The algorithm computed by the UNIX ``cksum'' command, as defined by the Single UNIX Specification, Version 2

So, it seems fairly natural to call this digest algorithm `cksum`.  At https://reveng.sourceforge.io/crc-catalogue/all.htm, there are 12 different CRC-32 algorithms documented, so it seems reasonable to name the `cksum` variant this.

In terms of compatibility, at least the xrootd-hdfs plugin and xrootd-multiuser plugin utilize the `cksum` name for the digest algorithm name for this particular algorithm.

In the "bad news" department, it appears the built-in algorithms call the cksum algorithm the `crc32` digest whereas xrootd-hdfs/multiuser call the Zlib CRC32 algorithm the `crc32` digest (which the built-in algorithms call `zcrc32`.  That means someone requesting the `crc32` checksum algorithm will get differing results, which is really unfortunate.

Reply:
The only things I know about cksum is that we don't support natively that checksum and that it should not be base64 encoded before being returned by the client.

`addChecksumToMaps(std::make_unique<XrdHttpChecksum>("cksum","UNIXcksum",false));`

So it's up to the person writing the script to deal with what the return value should be isn't it?

Reply:
@bbockelm thanks for your reply. What you said about the xrootd-hdfs plugins triggers some questioning on my side: I wanted to be nice with server administrator allowing them to set, for example, `sha512` or `sha-512` for configuring a sha-512 algorithm. This would not have worked before my modification, see this function here: https://github.com/xrootd/xrootd/blob/master/src/XrdHttp/XrdHttpReq.cc#L134

If there are other plugins that can be used to compute checksums, could my new implementation cause problems?

Reply:
For xrootd checksum plugins the return value is always a hex-string and that implies for certain checksums like SHAxxx (which we don't have implementation for as of yet) the HTTP code would need to convert it to base64. Thus far, all script implementation do the same. Having a common format for all checksums is the "standard" for xroot but not for http. I will make that obvious in the documentation. I think the table already specifies when conversion is necessary. The only odd one is Unixcksum (a.k.a. cksum) which is supposed to be a dec-string. However, since you are removing from the table the conversion issue becomes moot.